lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -10,6 +10,18 @@
 	die( 'Silence is golden.' );
 }
 
+/**
+ * Registers the extension of the WP_REST_Posts_Controller class,
+ * to add support for post formats.
+ * - Hardcoded the 'post' post type for now.
+ */
+function gutenberg_post_format_rest_posts_controller() {
+	$posts_controller = new Gutenberg_REST_Posts_Controller_6_7( 'post' );
+	$posts_controller->register_routes();
+}
+
+add_action( 'rest_api_init', 'gutenberg_post_format_rest_posts_controller' );
```

Comment:
I think the preferred way of doing this is using the `register_post_type_args` filter.

```php
function gutenberg_post_format_rest_posts_controller( $args, $post_type ) {
	if ( 'post' === $post_type ) {
		$args['rest_controller_class'] = 'Gutenberg_REST_Posts_Controller_6_7';
	}

	return $args;
}
add_filter( 'register_post_type_args', 'gutenberg_post_format_rest_posts_controller', 10, 2 );
```

Reply:
I don't know, so yes, I am open to changing it.
The question remains how to support post formats for custom post types.

Reply:
Good questions! We could change the check to something like this:

```php
function gutenberg_post_format_rest_posts_controller( $args, $post_type ) {
	if ( post_type_supports( $post_type, 'post-formats' ) ) {
		$args['rest_controller_class'] = 'Gutenberg_REST_Posts_Controller_6_7';
	}

	return $args;
}
add_filter( 'register_post_type_args', 'gutenberg_post_format_rest_posts_controller', 10, 2 );
```